src/plugins/controller/UnicyclePID/UnicyclePID.cpp

```diff
@@ -93,33 +94,39 @@ void UnicyclePID::init(std::map<std::string, std::string> &params) {
     if (!speed_pid_.init(params["speed_pid"], false)) {
         std::cout << "Failed to set speed PID" << std::endl;
     }
+
+    parent_->state() = std::make_shared<State>();
+    *(parent_->state()) = *(parent_->state_truth());
```

Comment:
The controller shouldn't be acting on the state's "truth," but a possibly noisy version of the truth since this representative of a real system. The sensor plugin interface handles setting the controller plugin's "state_" variable to be the "truth" or "noisy" versions depending on whether the NoisyState plugin was used or not.

Reply:
Agreed.  Done.